diff --git a/CMake/ConfigureBuild.cmake b/CMake/ConfigureBuild.cmake
index 0e04f59701..40f933c4bc 100644
--- a/CMake/ConfigureBuild.cmake
+++ b/CMake/ConfigureBuild.cmake
@@ -149,7 +149,7 @@ if (BUILD_TYPE_LOWER STREQUAL "debug" AND WIN32)
 endif ()
 
 # Create the pkg-config package files on Unix systems
-if (UNIX)
+#if (UNIX)
   set(OGRE_PLUGIN_PREFIX "")
   set(OGRE_PLUGIN_EXT ".so")
   set(OGRE_PAGING_ADDITIONAL_PACKAGES "")
@@ -158,12 +158,17 @@ if (UNIX)
     set(OGRE_PLUGIN_EXT ".a")
   endif ()
 
+  if (WIN32)
+    set(OGRE_PLUGIN_PREFIX "")
+    set(OGRE_PLUGIN_EXT ".lib")
+  endif()
+
   set(OGRE_ADDITIONAL_LIBS "")
   set(OGRE_ADDITIONAL_INCLUDE_DIRS "")
 
   set(OGRE_CFLAGS "")
-  set(OGRE_PREFIX_PATH ${CMAKE_INSTALL_PREFIX})
-  if (OGRE_CONFIG_THREADS GREATER 0)
+  set(OGRE_PREFIX_PATH "")
+  if (UNIX AND OGRE_CONFIG_THREADS GREATER 0)
     set(OGRE_CFLAGS "-pthread")
     set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lpthread")
   endif ()
@@ -236,7 +241,7 @@ if (UNIX)
     configure_file(${OGRE_TEMPLATES_DIR}/OGRE-HLMS.pc.in ${PROJECT_BINARY_DIR}/pkgconfig/OGRE-HLMS.pc @ONLY)
     install(FILES ${PROJECT_BINARY_DIR}/pkgconfig/OGRE-HLMS.pc DESTINATION ${OGRE_LIB_DIRECTORY}/pkgconfig)
   endif ()
-endif ()
+# endif ()
 
 if(OGRE_CONFIG_STATIC_LINK_CRT)
 #We statically link to reduce dependencies
diff --git a/CMake/Templates/OGRE.pc.in b/CMake/Templates/OGRE.pc.in
index 5d752f2991..a4baaafa65 100644
--- a/CMake/Templates/OGRE.pc.in
+++ b/CMake/Templates/OGRE.pc.in
@@ -2,7 +2,7 @@ prefix=@OGRE_PREFIX_PATH@
 exec_prefix=${prefix}
 libdir=${prefix}/@OGRE_LIB_DIRECTORY@
 includedir=${prefix}/include
-plugindir=${libdir}/OGRE
+plugindir=${prefix}/@OGRE_PLUGIN_DIRECTORY@
 
 Name: OGRE
 Description: Object-Oriented Graphics Rendering Engine
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 376af45b89..bcdab0b274 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -388,6 +388,12 @@ cmake_dependent_option(OGRE_CONFIG_STATIC_LINK_CRT "Statically link the MS CRT d
 set(OGRE_LIB_DIRECTORY "lib${LIB_SUFFIX}" CACHE STRING "Install path for libraries, e.g. 'lib64' on some 64-bit Linux distros.")
 set(OGRE_BIN_DIRECTORY "bin" CACHE STRING "Install path for binaries")
 
+if (WIN32)
+  set(OGRE_PLUGIN_DIRECTORY "bin" CACHE STRING "Install path for plugins.")
+else()
+  set(OGRE_PLUGIN_DIRECTORY "lib${LIB_SUFFIX}/OGRE" CACHE STRING "Install path for plugins.")
+endif()
+
 cmake_dependent_option(OGRE_INSTALL_VSPROPS "Install Visual Studio Property Page." FALSE "MSVC" FALSE)
 if (OGRE_INSTALL_VSPROPS)
   configure_file(${OGRE_TEMPLATES_DIR}/OGRE.props.in ${PROJECT_BINARY_DIR}/OGRE.props)
