diff --git a/Components/RTShaderSystem/CMakeLists.txt b/Components/RTShaderSystem/CMakeLists.txt
index dff7476ef..628714cfc 100644
--- a/Components/RTShaderSystem/CMakeLists.txt
+++ b/Components/RTShaderSystem/CMakeLists.txt
@@ -16,7 +16,7 @@ list(APPEND HEADER_FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/include/OgreRTShader.i)
 file(GLOB SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp")
 
-get_native_precompiled_header(OgreRTShaderSystem)
+# get_native_precompiled_header(OgreRTShaderSystem)
 add_library(OgreRTShaderSystem ${OGRE_COMP_LIB_TYPE} ${OgreRTShaderSystem_pch} ${HEADER_FILES} ${SOURCE_FILES})
 set_target_properties(OgreRTShaderSystem PROPERTIES VERSION ${OGRE_SOVERSION} SOVERSION ${OGRE_SOVERSION})
 target_link_libraries(OgreRTShaderSystem PUBLIC OgreMain)
@@ -26,7 +26,7 @@ target_include_directories(OgreRTShaderSystem PUBLIC
 
 # repeat for PCH
 include_directories("${PROJECT_SOURCE_DIR}/OgreMain/include" "${PROJECT_BINARY_DIR}/include" "${CMAKE_CURRENT_SOURCE_DIR}/include" "${CMAKE_CURRENT_SOURCE_DIR}/src")
-add_native_precompiled_header(OgreRTShaderSystem "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreShaderPrecompiledHeaders.h")
+# add_native_precompiled_header(OgreRTShaderSystem "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreShaderPrecompiledHeaders.h")
 
 generate_export_header(OgreRTShaderSystem 
     EXPORT_MACRO_NAME _OgreRTSSExport
diff --git a/OgreMain/CMakeLists.txt b/OgreMain/CMakeLists.txt
index a0d3f1852..c676df921 100644
--- a/OgreMain/CMakeLists.txt
+++ b/OgreMain/CMakeLists.txt
@@ -189,7 +189,7 @@ set (TARGET_LINK_FLAGS "")
 
 # setup OgreMain target
 
-get_native_precompiled_header(OgreMain)
+# get_native_precompiled_header(OgreMain)
 add_library(OgreMain ${OGRE_LIB_TYPE} ${OgreMain_pch} ${HEADER_FILES} ${SOURCE_FILES} ${PLATFORM_HEADERS} ${PLATFORM_SOURCE_FILES} ${THREAD_HEADER_FILES} ${THREAD_SOURCE_FILES})
 
 if (WINDOWS_STORE OR WINDOWS_PHONE)
@@ -264,7 +264,7 @@ endif ()
 target_link_libraries(OgreMain PUBLIC ${PLATFORM_LIBS} PRIVATE ${LIBRARIES} ${CMAKE_DL_LIBS})
 
 # specify a precompiled header to use
-add_native_precompiled_header(OgreMain "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreStableHeaders.h")
+# add_native_precompiled_header(OgreMain "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreStableHeaders.h")
 
 # install OgreMain
 ogre_config_lib(OgreMain TRUE)
