diff --git a/OgreMain/CMakeLists.txt b/OgreMain/CMakeLists.txt
index 883aa2f5a..918f102ed 100644
--- a/OgreMain/CMakeLists.txt
+++ b/OgreMain/CMakeLists.txt
@@ -54,8 +54,8 @@ elseif (APPLE)
   else ()
     include_directories("${CMAKE_CURRENT_SOURCE_DIR}/src/OSX")
     file(GLOB PLATFORM_SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/OSX/*.cpp"
-      "${CMAKE_CURRENT_SOURCE_DIR}/src/OSX/*.mm"
-    )
+      "${CMAKE_CURRENT_SOURCE_DIR}/src/OSX/*.mm")
+    set(PLATFORM_LIBS "-framework CoreFoundation -framework Foundation")
   endif ()
 elseif(ANDROID)
   # required by OgrePlatformInformation.cpp

